### DISCUSSION TITLE
Type-only support for standalone Typescript frontend?

### DISCUSSION THREAD

**Discussion by thw0rted** [prisma/prisma#17567] [Q&A]
I have a question that's kind of similar to https://github.com/prisma/prisma/discussions/15566, but instead of having multiple microservices talking to the same database, we have a backend that uses Prisma to load business objects from the database and sends them to both a separate (Angular) frontend, as well as other services.  It's great how Prisma provides robust type definitions for these business objects, but now I want to share the generated type information (only!) with these other projects.

I found [this older blog post](https://www.prisma.io/blog/full-stack-typesafety-with-angular-nest-nx-and-prisma-CcMK7fbQfTWc) about sharing types across the stack, but it recommends several additional layers of generators and frameworks that we aren't using, and assumes the whole stack is in a monorepo (again, not using).

My first impulse was to run `prisma generate` on the backend, then copy out the created `node_modules/.prisma/client/index.d.ts` into its own NPM package that the other consumers can read.  Of course this will have to be manually kept up to date when our `schema.prisma` changes, but I was trying to avoid having every consumer stuck with an additional dependency on Prisma (and an additional build step, etc), when none of them actually interact with the database.

Is there first class support for something like `prisma generate-types-only`?  If not, is there enough interest to open an issue?

**Comment by nurul3101** [comment]
Hello @thw0rted 👋 

As you mentioned, the solution is to create an npm package to share the types across multiple repositories.
And you are correct, you would need to upgrade the package as your schema changes. 

Do you have any recommendations on how this could be avoided, you would still need to update the npm package as schema evolves, right? The generated types are always updated when you invoke the `npx prisma generate` command.

**Reply by thw0rted** [reply]
I must admit I'm still pretty new with Prisma.  I guess I was hoping for a something that would cleanly generate "model only" type information, without constructors and CRUD functions.  Currently, generated types start with an import from `@prisma/client/runtime`, so I think consumers will still need a dev dependency on `@prisma/client` to avoid typechecker errors, even though as the name suggests I think these types are only actually used for model constructors, query-time error handling, etc.

It sounds like maybe what I'm doing now will get the job done, but it might be worthwhile for me to open an issue to talk about model-only generation.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
